bin/import_xml2obd.py

```diff
@@ -157,21 +185,105 @@ def get_preset(name):
     return None
 
 
+def upload_resources(remote_site, api_key, jsonfile, resource_directory):
+    #remote site
+    site = ckanapi.RemoteCKAN(
+        remote_site,
+        apikey=api_key,
+        user_agent='ckanapi-uploader/1.0')
+
+    #load dataset and resource file to cloud
+    ds = read_json(jsonfile)
+    for rec in ds:
+        try:
+            #site.action.package_delete(id=rec['id'])
+            #return
+            #site.action.package_create(**rec)
+            site.action.package_update(**rec)
+        except ckan.logic.NotFound:
+            try:
+                site.action.package_create(**rec)
+            except ckan.logic.ValidationError:
+                traceback.print_exc()
+                print(rec)
+                continue
+        except ckan.logic.ValidationError:
+            traceback.print_exc()
+            print(rec)
+            continue
+        res = rec['resources'][0]
+        source = resource_directory + '/' + res['url'].split('/')[-1]
+        with open(source) as f:
+            try:
+                rc = site.action.resource_patch(
+                    id=res['id'],
+                    upload=(os.path.basename(source), f))
+            except ckanapi.errors.CKANAPIError:
+                traceback.print_exc()
+                print(rec)
+                print("failed upload " + os.path.basename(source))
+                continue
+
+        print("updated for " + os.path.basename(source))
+
+
 def main():
     global presets, audience, canada_resource_type,canada_subject
     global canada_resource_language
+    canada_resource_format = None
+
+    # upload the generated json to remote site and upload the file
+    if sys.argv[1] =='upload':
+        return upload_resources(*sys.argv[2:])
+
     refs, extras = read_xml(sys.argv[1])
     read_organizations_from_json(sys.argv[2])
-    presets = read_presets(sys.argv[3])
+    presets = read_presets(sys.argv[3])  # TODO: read from doc.yaml
```

Comment:
`site.action.scheming_dataset_schema_show(type='doc')` is all you should need here